### PR TITLE
fix(idp-extraction-connector): fixed idp unstructured extraction malformed json response

### PR DESCRIPTION
**Pull request by reiballa**
## Description

This pull request introduces significant improvements to the Vertex AI Gemini integration by migrating the implementation to use the LangChain4j library, enhancing document content handling, and improving the robustness of AI response parsing. It also removes the use of the `maxTokens` parameter from several model callers to align with updated API usage. The changes are well-covered by new and updated unit tests.

**Vertex AI Gemini Integration & Document Handling:**
- Migrated the `VertexCaller` implementation to use LangChain4j's `VertexAiGeminiChatModel`, enabling native support for multimodal documents (PDFs, images, text, etc.) and simplifying authentication and system prompt handling. The new implementation also introduces a robust method for converting documents to the appropriate LangChain4j content types, supporting a wider range of file formats and providing sensible fallbacks. [[1]](diffhunk://#diff-307e120719d69079e2c2a8b8acfa8dcdfc08610b37599ea7b9fb8d43010b6803L9-L26) [[2]](diffhunk://#diff-307e120719d69079e2c2a8b8acfa8dcdfc08610b37599ea7b9fb8d43010b6803L38-R145)
- Added the `langchain4j-vertex-ai-gemini` dependency to the Maven configuration.

**Model Parameter Updates:**
- Removed the use of the `maxTokens` parameter from Bedrock and Azure AI Foundry callers, as well as associated test assertions, to reflect updated model API requirements. [[1]](diffhunk://#diff-85e2058c14d046fc330a3fa4e1f373f0763e5bbab6fb26b3ccae3c636d13dda5L65-R65) [[2]](diffhunk://#diff-da05aa8475df471eca18e4658ceef06238f10d6e2424873f1dee6ca061e01852L69) [[3]](diffhunk://#diff-da05aa8475df471eca18e4658ceef06238f10d6e2424873f1dee6ca061e01852L98) [[4]](diffhunk://#diff-5d8c95a240ac66a0b590f7ef095871671dd95458733f20e5a5bad8fe3bfbc686L147) [[5]](diffhunk://#diff-5d8c95a240ac66a0b590f7ef095871671dd95458733f20e5a5bad8fe3bfbc686L218)
@sahilbhatoacamunda i did this because i found out that the incident that started this was caused by the maxtoken limit being too low. I decided to ommit it for now until we have a more robust solution.

**AI Response Parsing & Utility Improvements:**
- Improved the parsing of AI responses by introducing a new utility method, `filterThinkingContent`, which removes `<thinking>...</thinking>` sections from responses before further processing. This helps ensure that only relevant content is parsed as JSON. [[1]](diffhunk://#diff-fcde2abb9aa3978be0db860e5283760bb73824ccc871f024074bc60cce9b380cL201-R203) [[2]](diffhunk://#diff-7fd8b6459c88e525868e162642985b3872e8d2fc02e56b004d578e9646a58e22R46-R74)

**Testing Enhancements:**
- Rewrote and expanded unit tests for `VertexCaller` to verify correct content conversion for various document types and edge cases, ensuring robust handling of PDFs, images, text, unknown, and null content types.

**Summary of most important changes:**

*Vertex AI Gemini & Document Handling*
- Migrated `VertexCaller` to use LangChain4j's Gemini model, enabling native multimodal document support and simplifying prompt/system message handling. Added robust document-to-content conversion logic. [[1]](diffhunk://#diff-307e120719d69079e2c2a8b8acfa8dcdfc08610b37599ea7b9fb8d43010b6803L9-L26) [[2]](diffhunk://#diff-307e120719d69079e2c2a8b8acfa8dcdfc08610b37599ea7b9fb8d43010b6803L38-R145)
- Added `langchain4j-vertex-ai-gemini` dependency to `pom.xml`.

*Model Parameter Updates*
- Removed `maxTokens` usage from Bedrock and Azure AI Foundry callers and associated tests. [[1]](diffhunk://#diff-85e2058c14d046fc330a3fa4e1f373f0763e5bbab6fb26b3ccae3c636d13dda5L65-R65) [[2]](diffhunk://#diff-da05aa8475df471eca18e4658ceef06238f10d6e2424873f1dee6ca061e01852L69) [[3]](diffhunk://#diff-da05aa8475df471eca18e4658ceef06238f10d6e2424873f1dee6ca061e01852L98) [[4]](diffhunk://#diff-5d8c95a240ac66a0b590f7ef095871671dd95458733f20e5a5bad8fe3bfbc686L147) [[5]](diffhunk://#diff-5d8c95a240ac66a0b590f7ef095871671dd95458733f20e5a5bad8fe3bfbc686L218)

*AI Response Parsing*
- Added `StringUtil.filterThinkingContent` to remove `<thinking>` tags from AI responses before JSON parsing, improving robustness. [[1]](diffhunk://#diff-fcde2abb9aa3978be0db860e5283760bb73824ccc871f024074bc60cce9b380cL201-R203) [[2]](diffhunk://#diff-7fd8b6459c88e525868e162642985b3872e8d2fc02e56b004d578e9646a58e22R46-R74)

*Testing*
- Added and refactored tests for `VertexCaller` to cover all supported and fallback document content types.

## Related issues

closes https://github.com/camunda/connectors/issues/5383

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.
- [ ] Backport labels are added if these code changes should be backported. No backport label is added to the latest release, as this branch will be rebased onto main before the next release.

### CODE REVIEW

File: connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/caller/VertexCaller.java [hunk 1]
```diff
@@ -35,104 +35,113 @@ public String generateContent(ExtractionRequestData input, GcpProvider baseReque
...
+          chatModelBuilder.credentials(googleCredentials);
+        }
+      }
+
+      var chatModel = chatModelBuilder.build();
+
+      // Create user message with document content using LangChain4j's native document handling
+      String userMessageText = llmModel.getMessage(input.taxonomyItems());
+
+      List<dev.langchain4j.data.message.Content> messageContents = new ArrayList<>();
```
Comment (sahilbhatoacamunda) [inline connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/caller/VertexCaller.java:63]: Can you remove the full qualified names and just import the class directly?

File: connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/caller/VertexCaller.java [hunk 2]
```diff
@@ -35,104 +35,113 @@ public String generateContent(ExtractionRequestData input, GcpProvider baseReque
...
+      messageContents.add(new TextContent(userMessageText));
+
+      // Convert the document to appropriate LangChain4j content based on content type
+      dev.langchain4j.data.message.Content documentContent =
```
Comment (sahilbhatoacamunda) [inline connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/caller/VertexCaller.java:67]: Same here

File: connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/caller/VertexCaller.java [hunk 3]
```diff
@@ -35,104 +35,113 @@ public String generateContent(ExtractionRequestData input, GcpProvider baseReque
...
-      LOGGER.debug("Gemini generate content response: {}", output);
+      ChatResponse response;
+      if (llmModel.isSystemPromptAllowed()) {
+        response =
+            chatModel.chat(List.of(SystemMessage.from(llmModel.getSystemPrompt()), userMessage));
+      } else {
+        // For models that don't support system prompts, combine system and user messages
+        String combinedMessage =
+            String.format("%s\n\n%s", llmModel.getSystemPrompt(), userMessageText);
+        List<dev.langchain4j.data.message.Content> combinedContents = new ArrayList<>();
```
Comment (sahilbhatoacamunda) [inline connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/caller/VertexCaller.java:83]: Same here

**Review by sahilbhatoacamunda — APPROVED** [review]
I have left minor comments but approving it to unblock.
